File: evidence/generated-systems/golang/hr-on-golang/hr-on-golang/internal/model/CompensationPackage.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// CompensationPackage Declaration
//==============================================================
type CompensationPackage struct {
    gorm.Model
     EffectiveFrom                                                            time.Time
    EffectiveTo                                                            time.Time
    Currency                                    string
    ContractId         *uint
    Contract           *EmploymentContract `gorm:"foreignKey:ContractId"`
     SalaryComponents           []SalaryComponent `gorm:"foreignKey:SalaryComponentsFromCompensationPackageId"`
     BonusPlans           []BonusPlan `gorm:"foreignKey:BonusPlansFromCompensationPackageId"`
     EquityGrants           []EquityGrant `gorm:"foreignKey:EquityGrantsFromCompensationPackageId"`

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/hr-on-rails/hronrails/app/models/compensationpackage.rb
class CompensationPackage < ApplicationRecord


  has_many :Contract, class_name: 'EmploymentContract'
  has_many :SalaryComponents, class_name: 'SalaryComponent'
  has_many :BonusPlans, class_name: 'BonusPlan'
  has_many :EquityGrants, class_name: 'EquityGrant'

end
